servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/ServerContext.java

```diff
@@ -50,6 +50,35 @@ default void awaitShutdown() {
         awaitTermination(onClose().toFuture());
     }
 
+    /**
+     * Toggles the server's channel accepting connection ability.
+     * <p>
+     * Passing a {@code false} value, will stop accepting connections on the server channel, without affecting any other
+     * interaction to currently open child channels (i.e., reads / writes).
+     * <p>
+     * Depending on the transport, connections may still get ESTABLISHED, see.
+     * {@link ServiceTalkSocketOptions#SO_BACKLOG backlog} and (in case of Linux)
+     * <a href="https://www.kernel.org/doc/Documentation/networking/ip-sysctl.txt">SOMAXCONN</a>.
+     * For instance, in case of TCP the 3-way handshake may finish, and the connection will await in the
+     * accept queue to be accepted. If the accept queue is full, connection SYNs will await in the
+     * SYN backlog (i.e., in case of Linux <a href="https://www.kernel.org/doc/Documentation/networking/ip-sysctl.txt">
+     * tcp_max_syn_backlog</a>). These additional parameters may affect the behavior of new flows when the service
+     * is not accepting.
+     * <p>
+     * Depending on how long this stays in the {@code false} state, it may affect other timeouts (i.e., connect-timeout
```

Comment:
It sounds like this may make the behaviour worse for clients since they might wait until connect timeout to learn that the server is not accepting connections. Is it possible to adjust the backlog queues to zero while not accepting connections and reject the queued connections as if the queue had been full?

Not being able to fully reject the inbound connection seems bad.

Reply:
Good observation. That indeed may be a problem, and this is the reason I am trying to document all expectations in the javadoc so users can make an educated decision. I want to avoid making a global decision here, and give the tools to the user's to tune accordingly.

Reply:
Excellent point. I was wondering whether it makes more sense to not expose such capability if we can't provide an extensive guideline how to configure the TCP settings. Wouldn't a mechanism for accepting the connection and immediately replying with a 503 HTTP error code be more "by the book"? For gRPC that's also safe as the clients should honor that code according to this table: https://grpc.github.io/grpc/core/md_doc_http-grpc-status-mapping.html